kalman_watch/monitor_win_perigee.py

```diff
@@ -0,0 +1,768 @@
+    times_list = []
+    n_drops_list = []
+    colors_list = []
+    for idx, perigee_time in enumerate(perigee_times):
+        dat = fetch.MSIDset(
+            msids, perigee_time - duration * u.min, perigee_time + duration * u.min
+        )
+        if len(dat["aokalstr"]) > 200:
+            times, n_drops = get_binned_drops_from_npnt_tlm(dat)
+            times_list.append(times - perigee_time.secs)
+            n_drops_list.append(n_drops)
+            colors_list.append([f"C{idx}"] * len(times))
+
+    return KalmanDropsData(
+        start,
+        stop,
+        np.concatenate(times_list),
+        np.concatenate(n_drops_list),
+        np.concatenate(colors_list),
+    )
+
+
+def plot_kalman_drops(
+    kalman_drops_data: KalmanDropsData,
+    ax,
+    alpha: float = 1.0,
+    title: str | None = None,
+    marker_size: float = 10,
+) -> matplotlib.collections.PathCollection:
+    """Plot the number of kalman drops per minute.
+
+    Parameters
+    ----------
+    kalman_drops_data : KalmanDropsData
+        Output of get_kalman_drops_nman or get_kalman_drops_npnt
+    ax : matplotlib.axes.Axes
+        Matplotlib axes
+    alpha : float
+        Alpha value for scatter plot
+    title : str, None
+        Title for plot (default=None)
+    marker_size : float
+        Marker size for scatter plot (default=10)
+
+    Returns
+    -------
+    scat : matplotlib.collections.PathCollection
+        Scatter plot collection
+    """
+    scat = ax.scatter(
+        kalman_drops_data.times / 60,
+        kalman_drops_data.kalman_drops.clip(None, 160),
+        s=marker_size,
+        c=kalman_drops_data.colors,
+        alpha=alpha,
+    )
+    # set major ticks every 10 minutes
+    ax.xaxis.set_major_locator(plt.MultipleLocator(10))
+    if title is None:
+        title = (
+            f"Kalman drops per minute near perigee {kalman_drops_data.start.iso[:7]}"
+        )
+    if title:
+        ax.set_title(title)
+    ax.set_xlabel("Time from perigee (minutes)")
+    return scat
+
+
+def plot_mon_win_and_aokalstr_composite(
+    kalman_drops_npnt, kalman_drops_nman_list, outfile=None, title=""
+):
+    """Plot the monitor window and kalman drops data.
+
+    Parameters
+    ----------
+    kalman_drops_npnt : KalmanDropsData
+        Output of get_kalman_drops_npnt
+    kalman_drops_nman_list : list[KalmanDropsData]
+        Output of get_kalman_drops_nman
+    outfile : str
+        Output file name (default=None)
+    title : str
+        Title for plot (default="")
+
+    Returns
+    -------
+    None
+    """
+    fig, ax = plt.subplots(1, 1, figsize=(10, 3))
+
+    plot_kalman_drops(
+        kalman_drops_npnt,
+        ax=ax,
+        alpha=0.4,
+    )
+
+    for mon_win_kalman_drops in kalman_drops_nman_list:
+        plot_kalman_drops(
+            mon_win_kalman_drops, ax=ax, alpha=1.0, marker_size=20, title=""
+        )
+
+    ax.set_title(title)
+    fig.tight_layout()
+
+    if outfile:
+        fig.savefig(outfile)
```

Comment:
I think the default dpi of 100 is pretty blurry.

Reply:
Looks fine to me but maybe my vision isn't sharp enough. How many dpi do you want?

Reply:
I leave it to you to decide if it is worth the work -- for me 150 or 200 is good on these retina displays.  I'd probably go with 200 and limit the width to 1000px in the html.

Reply:
Done.